message_app/send_message.py

```diff
@@ -160,10 +159,27 @@ def message_handler(data):
     emit("message_handler_client", return_data, room=conversation_id)
 
 # A socket that updates user's last_active_time
-@socketio.on("last_active", namespace="/messages")
-def last_active(data):
+@socketio.on("last_active_time", namespace="/messages")
+def last_active_time(data):
+    # The data is a dictionary with three keys:
+    # username: username
+    # last_active_time: new last_active_time to update in the database
+    # is_closing: a boolean key that track if the user close their browser
     username = data["username"]
     last_active_time = data["last_active_time"]
+    is_closing = data["is_closing"]
     current_user = User.select(username)
+    old_status = check_user_status(current_user.last_active_time)
+    new_status = "away" if is_closing else check_user_status(last_active_time)
```

Comment:
If the user closed their browser, their status will always change to **_away_** without checking their last_active_time.